compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -343,6 +343,7 @@ object Checking {
         fail(i"$sym cannot have the same name as ${cls.showLocated} -- class definitions cannot be overridden")
         sym.setFlag(Private) // break the overriding relationship by making sym Private
       }
+    avoidPrivateLeaks(sym, sym.pos)
```

Comment:
So far, no method in `Checking` mutates anything, it would be nice to keep this pattern. I suggest doing what `checkNonCyclic` does and returning the potentially updated `sym.info`, i.e.:
```scala
if (!sym.is(SyntheticOrPrivate) && sym.owner.isClass)
  checkNoPrivateLeaks(sym, sym.pos)
else
  sym.info
```